src/Debugging/Debugger.php

```diff
@@ -0,0 +1,226 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Saloon\Debugging;
+
+use Saloon\Contracts\DebuggingDriver;
+use Saloon\Debugging\Drivers\RayDebugger;
+use Saloon\Exceptions\UnknownDriverException;
+use Saloon\Debugging\Drivers\ErrorLogDebugger;
+use Saloon\Debugging\Drivers\SystemLogDebugger;
+
+class Debugger
+{
+    /**
+     * Globally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected static array $globalRegisteredDrivers = [];
+
+    /**
+     * Locally registered drivers
+     *
+     * @var array<string, \Saloon\Contracts\DebuggingDriver>
+     */
+    protected array $registeredDrivers = [];
+
+    /**
+     * Drivers that have been subscribed to
+     *
+     * @var array<string, bool>
+     */
+    protected array $useDrivers = [];
+
+    /**
+     * Denotes if we send the request to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showRequest = false;
+
+    /**
+     * Denotes if we send the response to the debugging driver
+     *
+     * @var bool
+     */
+    protected bool $showResponse = false;
+
+    public function __construct()
+    {
+        $this->registerDriver(new RayDebugger);
+        $this->registerDriver(new ErrorLogDebugger);
+        $this->registerDriver(new SystemLogDebugger);
+    }
+
+    /**
+     * Register a driver globally
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return static
+     */
+    public static function registerGlobalDriver(DebuggingDriver $driver): static
+    {
+        static::$globalRegisteredDrivers[$driver->name()] = $driver;
+
+        return new static;
+    }
+
+    /**
+     * Register a driver
+     *
+     * @param \Saloon\Contracts\DebuggingDriver $driver
+     *
+     * @return $this
+     */
+    public function registerDriver(DebuggingDriver $driver): static
+    {
+        $this->registeredDrivers[$driver->name()] = $driver;
+
+        return $this;
+    }
+
```

Comment:
In case there are very niched configs that only applies to a specific driver, it might be useful to be able to individually configure them.
Primarily thinking of the syslog and error log drivers here.
If we're going the JSON route that @Sammyjo20 suggested (which is better, imo), then it'd be nice to be able to set JSON config (like pretty print).

```suggestion

    /**
     * Configure an already registered driver.
     *
     * @param string $driverName The name of a registered DebuggingDriver.
     * @param callable(\Saloon\Contracts\DebuggingDriver): (void)|null $callback
     *
     * @return $this
     * @throws \Saloon\Exceptions\UnknownDriverException
     */
    public function configureDriver(string $driverName, callable $callback): static
    {
        $registeredDrivers = $this->getRegisteredDrivers();

        $driver = $registeredDrivers[$driverName] ?? throw new UnknownDriverException(sprintf(
            'Unable to find the "%s" driver. Registered drivers: %s',
            $driverName,
            implode(', ', array_keys($registeredDrivers)),
        ));
        
        $callback($driver);
        
        return $this;
    }

```